DotNet UWP Samples/Earthquake Simulator/Earthquake Simulator/MainPage.xaml.cs

```diff
@@ -0,0 +1,142 @@
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+// The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=402352&clcid=0x409
+
+namespace Microsoft.DataStreamer.Samples.EarthquakeSimulator
+{
+    public sealed partial class MainPage : Page
+    {
+        // default state is to send data
+
+        public MainPage()
+        {
+            this.InitializeComponent();
+
+            this.DataContext = App.Instance.ViewModel;
+            App.Instance.Dispatcher = this.Dispatcher;
+        }
+
+        private void ButtonStartData_Click(object sender, RoutedEventArgs e)
+        {
+            App.Instance.StartEarthquake();
+        }
+
+        private void ButtonStopData_Click(object sender, RoutedEventArgs e)
+        {
+            App.Instance.StopEarthquake();
+        }
+
+        private async void ButtonStartRecording_Click(object sender, RoutedEventArgs e)
+        {
+            try
+            { 
+                var docs = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
+
+                await App.StreamerService.StartRecording(Path.Combine(docs, "DataStreamer", "Earthquake.csv"));
+            }
+            catch(Exception ex)
+            {
+                await Task.CompletedTask;
+            }
+        }
+
+        private async void ButtonStopRecording_Click(object sender, RoutedEventArgs e)
+        {
+            try
+            { 
+                await App.StreamerService.StopRecording();
+            }
+            catch(Exception ex)
+            {
+                await Task.CompletedTask;
+            } 
+        }
+
+        private async void ButtonReset_Click(object sender, RoutedEventArgs e)
+        {
+            try
+            { 
+                await App.StreamerService.Reset();
+            }
+            catch(Exception ex)
+            {
+                await Task.CompletedTask;
+            }
+        }
+
+        private async void ButtonClearOutput_Click(object sender, RoutedEventArgs e)
+        {
+            try
+            { 
+                await App.Instance.ViewModel.SetOutputLines("");
+            }
+            catch(Exception ex)
+            {
+                await Task.CompletedTask;
+            }  
+        }
+        
+        private async void ButtonReady_Click(object sender, RoutedEventArgs e)
+        {
+            try
+            { 
+                await App.StreamerService.Ready();
+            }
+            catch(Exception ex)
+            {
+                await Task.CompletedTask;
+            }
+        }        
+        
+        private async void ButtonNotReady_Click(object sender, RoutedEventArgs e)
+        {
+            try
+            { 
+                await App.StreamerService.NotReady();
+            }
+            catch
+            {
+
+            }        
```

Comment:
why do we need a try/catch block here?

Reply:
Entry points should **always** have try catch blocks